### PR TITLE
Pull in azure customdata from the dvd that azure mounts

### PR DESCRIPTION
**Pull request by imbstack**
Rather than using the instance metadata service which seems to randomly [not pass through certain customdata strings](https://github.com/MicrosoftDocs/azure-docs/issues/30370). 

I have somewhat verified that this _should_ work by logging into windows/linux vms and checking with the customdata files that are there. I know this bit entirely works in linux. Somewhat guessing on windows for now.

I am in no way proud of this code, please feel free to tell me to scrap bits or rewrite whatever!

### CODE REVIEW

File: provider/azure/metadata_darwin.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// +build darwin
```
Comment (djmitche) [inline provider/azure/metadata_darwin.go]: This is implied by the filename (but doesn't hurt)

File: provider/azure/metadata_linux.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	XMLName                           xml.Name                          `xml:"ProvisioningSection"`
+	LinuxProvisioningConfigurationSet LinuxProvisioningConfigurationSet `xml:"LinuxProvisioningConfigurationSet"`
+}
+
+type LinuxProvisioningConfigurationSet struct {
+	XMLName    xml.Name `xml:"LinuxProvisioningConfigurationSet"`
+	CustomData string   `xml:"CustomData"`
+}
+
+var customDataPath = "/var/lib/waagent/ovf-env.xml"
```
Comment (djmitche) [inline provider/azure/metadata_linux.go]: Do we need to do anything special in the deployment of start-worker to ensure it runs after the windows azure agent?

File: provider/azure/metadata_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoadCustomData(t *testing.T) {
+
+	customDataPath = "testdata\\CustomData.bin"
```
Comment (djmitche) [inline provider/azure/metadata_windows_test.go]: It's probably good hygiene to reset this to its original value with a `defer` block..

File: provider/azure/metadata_linux.go (hunk 2 above)
Comment (milescrabill) [reply provider/azure/metadata_linux.go]: I think we can add a service dependency in NSSM like so `nssm set worker-runner DependOnService WindowsAzureGuestAgent`. I found the service name [here](https://docs.microsoft.com/en-us/archive/blogs/mast/vm-agent-available-for-azure-vms).

Comment (djmitche) [reply provider/azure/metadata_linux.go]: Sounds good.  There's a deployment doc in the docs directory that would be a good place to put that information.

**Comment by imbstack** [comment]
The test failed due to workers being broken. I retriggered it and it passed.

**Pull request merged**